File: F1App/F1App/Models/RaceEvent.swift
import Foundation

struct RaceEvent: Identifiable, Decodable, Hashable {
    let id: Int64
    let eventType: EventType
    let timestampMs: Int64
    let lap: Int?
    let driverNumber: Int?
    let driverNumberOvertaken: Int?
    let message: String?
    let subtype: String?
    let extra: [String: String]?

    enum EventType: String, Decodable {
        case overtake
        case race_control
    }
}

struct ToastEvent: Identifiable, Hashable {
    let id: Int64
    let event: RaceEvent
    let expiresAtMs: Int64
}

File: API/F1_API/app/Models/RaceEvent.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class RaceEvent extends Model
{
    use HasFactory;

    protected $fillable = [
        'session_key',
        'event_type',
        'timestamp_ms',
        'lap',
        'driver_number',
        'driver_number_overtaken',
        'message',
        'subtype',
        'extra',
    ];

    protected $casts = [
        'extra' => 'array',
    ];
}
